### ISSUE TITLE
trial design and trait data issue

### ISSUE THREAD

**Issue by isaak** [solgenomics/sgn#1720]
A trial with RCBD design entry shows on the trial detail page it has two blocks. But when you query for the phenotype data from the trial for certain traits (dry yield and dry matter content), you get only one block data. (Looks like these traits were not phenotyped in the second block??) When you get dataset of just the two traits, it is difficult to tell whether an entire block is missing data for the two traits or the RCBD is a mis-entry.

https://cassavabase.org/breeders/trial/102?format=

Similar issues: #1115, #1030

**Comment by bauchetg** [comment]
Hello Isaak, for https://cassavabase.org/breeders/trial/102?format= this is a layout format error, rep is defined as 1 and block as 2. they both should be 2 as it is a complete and balanced design. Data was collected for the two blocks.

**Comment by isaak** [comment]
Yes, for most traits I can see that data was collected for both blocks and it is an RCBD trial. The confusion is if you just happen to download data for only the traits that were not phenotyped on the second block. Or analysis based on RCBD model breaks for the two traits.

**Comment by nickmorales** [comment]
@isaak im not sure i get what the issue is here. is the phenotype data download not showing data you are expecting it to show? if a trait was only phenotyped on some plots, the phenotype data download will only return the plots that the trait was phenotyped on. there is a similar issue here https://github.com/solgenomics/sgn/issues/1593 that will be solved by returning all the plots in a trial even if they were not phenotyped for a triat (only in the case if the user queries for a specific trial e.g. does a phenotype download from the trial detail page).

**Comment by isaak** [comment]
@nickmorales: this issue is similar to #1593, indeed. The problem is: When a trial has a certain design, e.g. RCBD with 2 blocks, then when I query for a trait data from the trial, I expect data for plots from both blocks. If, somehow, one of the blocks was not phenotyped for a trait, it should show missing values for the trait. Otherwise, when you get trait data for one block only, then you wonder if the trial is indeed RCBD with 2 blocks. Programmatically, the consequence is that when you are running analysis expecting RCBD data, this breaks it. (I am adding more checks in my R scripts for this kind of anomaly). 

Again this problem is when you download for a single trait. If you download for all traits in the trial, it is obvious that the trial has RCBD with two blocks but some traits were not phenotyped for a block.

**Comment by isaak** [comment]
Resolved by displaying error message that describes the problem: There is only one sampling level and > 1 is required.
